amd/src/userlist_updater.js

```diff
@@ -67,34 +69,48 @@ define(['jquery', 'core/ajax'], function($, Ajax) {
-
             userselements.remove();
 
-            $(response).each(function(index, option) {
-                if (option.credential_url) {
-                    output += "<div id='fitem_id_certlink" + option.id + "' class='form-group row fitem'>";
-                    output += "<div class='col-md-3 col-form-label d-flex pb-0 pr-md-0'>";
-                    output += "<span class='d-inline-block'>" + option.name + "   " + option.email + "</span></div>";
-                    output += "<div class='col-md-9 form-inline align-items-start felement' data-fieldtype='static'>";
-                    output += "<div class='form-control-static'>Certificate " + option.credential_id;
-                    output += " - <a href='" + option.credential_url + "' target='_blank'>link</a></div></div></div>";
+            $(response).each(function(index, user) {
+                if (user.credential_url) {
+                  var context = {
+                    element: {
+                      html: 'Certificate ' + user.credential_id + ' - <a href='+ user.credential_url +' target="_blank">link</a>',
+                      staticlabel: true
+                    },
+                    label: user.name + '   ' + user.email
+                  };
                 } else {
-                    output += "<div class='form-group row fitem checkboxgroup1'>";
-                    output += "<div class='col-md-3'></div><div class='col-md-9 checkbox'>";
-                    output += "<div class='form-check d-flex'>";
-                    output += "<input type='hidden' name='users[" + option.id + "]' value='0'>";
-                    output += "<input type='checkbox' name='users[" + option.id + "]'"; 
-                    output += "class='form-check-input checkboxgroup1' value='1' id='id_users_" + option.id + "'>";
-                    output += "<label for='id_users_" + option.id + "'>";
-                    output += option.name + "   " + option.email;
-                    output += "</label></div></div></div>";
+                  var context = {
+                    element: {
+                      id: user.id,
+                      name: 'users['+ user.id +']',
+                      extraclasses: 'checkboxgroup1'
+                    },
+                    label: user.name + '   ' + user.email
+                  };
                 }
+
+                t.renderUser(context, user.credential_url);
             });
+        },
 
-            t.userscontainer.append(output);
+        /**
+         * Render the template with the user context.
+         *
+         * @param stdObject context - data for template.
+         * @param string certificate - certificate url to select correct template.
+         */
+        renderUser: function(context, certificate) {
+          template = certificate ? 'core_form/element-static' : 'core_form/element-advcheckbox';
+          
+          Templates.renderForPromise(template, context).then(function (_ref) {
```

Comment:
As @kei178 described in this [comment](https://github.com/accredible/moodle-mod_accredible/pull/61#discussion_r869943778), the `mod_form.php` will generate the form by rendering templates when doing `$mform->addElement`.

This function here will render the same templates that we are using in `mod_form` to display the users and then it will inject the rendered HTML in the page.

Docs about how to render a template with JS: https://docs.moodle.org/dev/Templates#Rendering_in_JavaScript